src/framework/piral-core/src/components/PiralRoutes.tsx

```diff
@@ -1,7 +1,19 @@
 import * as React from 'react';
 import { RouteComponentProps, SwitchProps } from 'react-router';
-import { useRoutes } from '../hooks';
-import { RouteSwitchProps } from '../types';
+import { useGlobalState } from '../hooks';
+import { RouteSwitchProps, AppPath } from '../types';
+import { useRouteFilter } from '../../app.codegen';
+
+function useRoutes() {
```

Comment:
Yes it was indeed, but since this is internal and we don't really need it I've moved it into the use of it. This way we don't have the otherwise cyclic relationship (and a few other benefits).